alkira/resource_alkira_connector_gcp_vpc_helper.go

```diff
@@ -1,39 +1,74 @@
 package alkira
 
 import (
-	"log"
-
 	"github.com/alkiranet/alkira-client-go/alkira"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func convertGcpRouting(in *schema.Set) *alkira.ConnectorGcpVpcRouting {
-	if in == nil || in.Len() > 1 {
-		log.Printf("[DEBUG] Only one object allowed in gcp routing options")
-		return nil
+func convertGcpRouting(in *schema.Set, subnets *schema.Set) *alkira.ConnectorGcpVpcRouting {
+	importOptions := alkira.ConnectorGcpVpcImportOptions{
+		RouteImportMode: "ADVERTISE_DEFAULT_ROUTE",
 	}
 
-	if in.Len() < 1 {
-		return nil
+	if in != nil && in.Len() == 1 {
+		for _, option := range in.List() {
+			cfg := option.(map[string]interface{})
+
+			if v, ok := cfg["prefix_list_ids"].([]interface{}); ok {
+				importOptions.PrefixListIds = convertTypeListToIntList(v)
+			}
+
+			if v, ok := cfg["custom_prefix"].(string); ok {
+				importOptions.RouteImportMode = v
+			}
+		}
+	}
+
+	exportAllSubnets := true
+
+	prefixes := generateGCPUserInputPrefixes(subnets)
+
+	if prefixes != nil && len(prefixes) > 0 {
+		exportAllSubnets = false
+	}
+
+	exportOptions := alkira.ConnectorGcpVpcExportOptions{
+		ExportAllSubnets: exportAllSubnets,
+		Prefixes:         prefixes,
 	}
 
 	gcp := &alkira.ConnectorGcpVpcRouting{
-		ImportOptions: alkira.ConnectorGcpVpcImportOptions{},
+		ExportOptions: exportOptions,
+		ImportOptions: importOptions,
 	}
+	return gcp
+}
 
-	for _, option := range in.List() {
-		cfg := option.(map[string]interface{})
+// generateUserInputPrefixes generate UserInputPrefixes used in GCP-VPC connector
+func generateGCPUserInputPrefixes(subnets *schema.Set) []alkira.UserInputPrefixes {
 
-		if v, ok := cfg["prefix_list_ids"].([]interface{}); ok {
-			gcp.ImportOptions.PrefixListIds = convertTypeListToIntList(v)
-		}
+	if subnets != nil && subnets.Len() > 0 {
 
-		if v, ok := cfg["custom_prefix"].(string); ok {
-			gcp.ImportOptions.RouteImportMode = v
+		prefixes := make([]alkira.UserInputPrefixes, subnets.Len())
+
+		for i, subnet := range subnets.List() {
+			r := alkira.UserInputPrefixes{}
+			t := subnet.(map[string]interface{})
+			if v, ok := t["id"].(string); ok {
+				r.Id = v
+			}
+			if v, ok := t["cidr"].(string); ok {
+				r.Value = v
+			}
+
+			r.Type = "SUBNET"
```

Comment:
Both Id and Value is optional, what if both are empty? Type will be still set to `SUBNET`... so some half empty struct will be sent to backend... which failed here many times before...

Reply:
added empty check in line 64 and tested.